ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -320,14 +321,21 @@ func (s *Service) updateService(ecsService *ecs.Service, newTaskDefinition *ecs.
 	oldTaskDefinitionId := entity.GetIdFromArn(ecsService.TaskDefinition)
 	newTaskDefinitionId := entity.GetIdFromArn(newTaskDefinition.TaskDefinitionArn)
 
+	if aws.StringValue(ecsService.SchedulingStrategy) == ecs.SchedulingStrategyDaemon {
+		count = nil
+		if oldTaskDefinitionId == newTaskDefinitionId {
+			newTaskDefinitionId = ""
```

Comment:
Although this results in a correct `UpdateService` call on L343, it means we still log the "Updated the ECS service with a new task definition..." msg on L348, which is not correct:
```
$~\amazon-ecs-cli> ./bin/local/ecs-cli.exe compose -f .\docker-compose.yml service up
?[36mINFO?[0m[0000] Using ECS task definition                     ?[36mTaskDefinition?[0m="685:1"
?[36mINFO?[0m[0000] Updated the ECS service with a new task definition. Old containers will be stopped automatically, and replaced with new ones  ?[36mdesiredCount?[0m=0 ?[36mforce-deployment?[0m=false ?[36mservice?[0m=685
?[36mINFO?[0m[0000] Service status                                ?[36mdesiredCount?[0m=3 ?[36mrunningCount?[0m=3 ?[36mserviceName?[0m=685
?[36mINFO?[0m[0000] ECS Service has reached a stable state        ?[36mdesiredCount?[0m=3 ?[36mrunningCount?[0m=3 ?[36mserviceName?[0m=685
```

Better to continue letting this case be handled by `updateServiceCount`, though the term "count" is admittedly misleading in this case (network & force-deployment updates will still apply).

Reply:
https://github.com/aws/amazon-ecs-cli/pull/685/commits/c6b01932fe468bfdce7fe4a6bec5d71b192c8970 corrects this.